File: MusicFestivalMPP/model/Ticket.cs
using System;

namespace MusicFestivalMPP.model;

public class Ticket: Entity
{
    public Performance Performance { get; set; }
    
    public String OwnerName { get; set; }
    
    public int NoOfSeats { get; set; }

    public Ticket(long id, Performance performance, String ownerName, int noOfSeats) : base(id)
    {
        Performance = performance;
        OwnerName = ownerName;
        NoOfSeats = noOfSeats;
    }

    public Ticket(Performance performance, String ownerName, int noOfSeats)
    {
        Performance = performance;
        OwnerName = ownerName;
        NoOfSeats = noOfSeats;
    }
}

File: code/src/main/java/model/Ticket.java
package model;

public class Ticket extends Entity {
    private Performance performance;
    private String ownerName;
    private Integer noOfSeats;

    public Ticket(Performance performance, String ownerName, Integer noOfSeats) {
        this.performance = performance;
        this.ownerName = ownerName;
        this.noOfSeats = noOfSeats;
    }

    public Ticket(Long ticketID, Performance performance, String ownerName, Integer noOfSeats) {
        setID(ticketID);
        this.performance = performance;
        this.ownerName = ownerName;
        this.noOfSeats = noOfSeats;
    }

    public Performance getPerformance() {
        return performance;
    }

    public void setPerformance(Performance performance) {
        this.performance = performance;
    }

    public String getOwnerName() {
        return ownerName;
    }

    public void setOwnerName(String ownerName) {
        this.ownerName = ownerName;
    }

    public Integer getNoOfSeats() {
        return noOfSeats;
    }

    public void setNoOfSeats(Integer noOfSeats) {
        this.noOfSeats = noOfSeats;
    }
}
